src/View/OpenPatientWindow.py

```diff
@@ -546,16 +546,17 @@ def on_loading_error(self, exception):
         """
         Error handling for progress window.
         """
+        errorMessage = ""
+                
         if type(exception[1]) == ImageLoading.NotRTSetError:
-            QMessageBox.about(self.progress_window, "Unable to open selection",
-                              "Selected files cannot be opened as they are not"
-                              " a DICOM-RT set.")
-            self.progress_window.close()
+            errorMessage = "Selected files cannot be opened as they are not a DICOM-RT set."
         elif type(exception[1]) == ImageLoading.NotAllowedClassError:
-            QMessageBox.about(self.progress_window, "Unable to open selection",
-                              "Selected files cannot be opened as they contain"
-                              " unsupported DICOM classes.")
-            self.progress_window.close()
+            errorMessage = "Selected files cannot be opened as they contain unsupported DICOM classes."
+        else:
+            errorMessage = "An error occurred while loading the selected files. Please try again."
```

Comment:
a user should only be encouraged to try again if this was a temporary error (and retrying might succeed).
The user might want to investigate the error and if they can correct it, try again.
If the data are missing StudyID... there might not be a fix (unless they want to do surgery on all of the DICOM objects missing the StudyID).
What is the intent/expectation here?

Reply:
What I've done is added a generic error handling here as previously there was no generic error handling. So if there was an uncaught error, Line: 556 would be displayed. This could definitely be refined further as I'm not entirely sure what you would like the application to behave if there's an uncaught exception.

Given if the StudyID is missing, this would've thrown `NotAllowedClassError` and Line:554 would've been displayed but I'm also not entirely sure if this is the correct intention and would further clarification.